distributed/utils_test.py

```diff
@@ -1800,8 +1812,22 @@ def check_instances():
     assert all(c.status == Status.closed for c in SpecCluster._instances), list(
         SpecCluster._instances
     )
-    SpecCluster._instances.clear()
+    wait_profiler()
+    gc.collect()
+    if Scheduler._instances:
+        s = next(iter(Scheduler._instances))
+        import objgraph
+
+        def ignore(obj):
+            return (
+                # ignore bound methods temporarily
+                not (inspect.ismethod(obj) and obj.__self__ is s))
 
+        objgraph.show_backrefs([s], filename="scheduler.png", filter=ignore)
```

Comment:
Otherwise, it seems to be `remove_worker` callback on the IOLoop and a `_check_lease_timeout` PeriodicCallBack that seem to be hanging on:
![scheduler](https://user-images.githubusercontent.com/3530212/166972496-7ad75948-9ab3-4484-b3b3-b74be0c33bf1.png)

Reply:
Placed the `check_instances` context outside of the `pristine_loop` context to solve this. Possibly controversial.